python/revng/cli/_commands/fetch_debuginfo/common.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python3
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import gzip
+import os
+import shutil
+import sys
+
+import requests
+
+
+def log(message, verbose):
+    if verbose:
+        sys.stderr.write(message + "\n")
+
+
+def download_file(url, local_filename, verbose):
+    log(f"Downloading {local_filename}...", verbose)
+    with requests.get(url, stream=True) as r:
+        if r.status_code == 200:
+            # TODO: Add support for other types of compressed files.
+            if r.headers["Content-Encoding"] == "gzip":
+                gzip_filename = local_filename + ".gz"
+                with gzip.open(gzip_filename, "wb") as gzip_file:
+                    shutil.copyfileobj(r.raw, gzip_file)
+                with gzip.open(gzip_filename, "rb") as gzip_file:
+                    content = gzip.decompress(gzip_file.read())
+                with open(local_filename, "wb") as debug_file:
+                    debug_file.write(content)
+                os.remove(gzip_filename)
+            else:
+                with open(local_filename, "wb") as debug_file:
+                    shutil.copyfileobj(r.raw, debug_file)
+
+            log("Downloaded...", verbose)
+            return True
+        else:
+            log("URL was not found...", verbose)
+            # Remove empty file.
+            os.remove(local_filename)
```

Comment:
Why remove if the file was never created?

Reply:
In some situations we end up with an empty file with the ```local_filename```.